### ISSUE TITLE
missing await and/or unused coroutine

### ISSUE THREAD

**Issue by pawciobiel** [pylint-dev/pylint#8870]
### Current problem

Currently pylint does not warn if there is a missing `await` / unused coroutine, eg:
```python
import asyncio

async def nested() -> None:
    print("first()")

async def fn() -> None:
    nested()  # <-  could this issue a warning please
    await asyncio.create_task(nested())
    await asyncio.gather(nested())
    await asyncio.ensure_future(nested())

async def main() -> None:
    await fn()

asyncio.run(main())
```

Slightly related to https://github.com/pylint-dev/pylint/issues/654

### Desired solution

It would be really cool if a call on awaitable without using future triggered a warning.

### Additional context

MS pyright: `error: Result of async function call is not used; use "await" or assign result to variable (reportUnusedCoroutine)`

mypy: 
 `error: Value of type "Coroutine[Any, Any, None]" must be used  [unused-coroutine]`
`note: Are you missing an await?`

**Comment by mbyrnepr2** [comment]
Looks like a duplicate of https://github.com/pylint-dev/pylint/issues/1496? We could keep this more recent issue and copy the example from that one to this ticket?

**Comment by mbyrnepr2** [comment]
Example from #1496:
```python
async def my_coro(*args):
    result = other_coro(*args)  # whoops, we didn't await
    x = result.some_attr  # error, coroutines don't have the some_attr attribute
    y = result["key"]  # error, coroutines don't have keys. The same applies to result[0]
    return result  # valid, coroutines can be returned as result sometimes.
```

**Comment by pawciobiel** [comment]
To me an assignment looks tricky because I think we would have to check
all possible operations on a coroutine/future not only in the current frame(?)
